ext4/dmverity/dmverity.go

```diff
@@ -0,0 +1,175 @@
+package dmverity
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+
+	"github.com/Microsoft/hcsshim/ext4/internal/compactext4"
+)
+
+const (
+	blockSize = compactext4.BlockSize
+)
+
+var (
+	salt                      = bytes.Repeat([]byte{0}, 32)
```

Comment:
hardcoded salt that is mentioned in the commit message

Reply:
Ideally we should keep all of the global error defs in one var block and move this right above

Reply:
```go
var salt = bytes.Repeat([]byte{0}, 32)

var (
        ErrSuperBlockReadFailure  = errors.New("failed to read dm-verity super block")
	ErrSuperBlockParseFailure = errors.New("failed to parse dm-verity super block")
	ErrRootHashReadFailure    = errors.New("failed to read dm-verity root hash")
)
```